pkg/catalog/actions_source.go

```diff
@@ -4,24 +4,51 @@ import (
 	"context"
 	"fmt"
 	"io"
+	"time"
 
 	"github.com/treeverse/lakefs/pkg/block"
+	"github.com/treeverse/lakefs/pkg/cache"
 	"github.com/treeverse/lakefs/pkg/graveler"
 )
 
 const repositoryLocation = "_lakefs_actions/"
 
 type ActionsSource struct {
 	catalog *Catalog
+	cache   cache.Cache
+}
+
+const (
+	DefaultActionsCacheSize   = 100
+	DefaultActionsCacheExpiry = 5 * time.Second
+	DefaultActionsCacheJitter = DefaultActionsCacheExpiry / 2
+)
+
+type ActionsCacheConfig struct {
+	Size   int
+	Expiry time.Duration
+	Jitter time.Duration
 }
 
 func NewActionsSource(catalog *Catalog) *ActionsSource {
 	return &ActionsSource{
 		catalog: catalog,
+		cache:   cache.NewCache(DefaultActionsCacheSize, DefaultActionsCacheExpiry, cache.NewJitterFn(DefaultActionsCacheJitter)),
 	}
 }
 
 func (s *ActionsSource) List(ctx context.Context, record graveler.HookRecord) ([]string, error) {
+	key := fmt.Sprintf("list_%s_%s", record.RepositoryID.String(), record.SourceRef.String())
```

Comment:
- Suggest using ':' as separator like we do for all cache keys.
- Don't think we require the prefix as the having <repo>:<ref> vs <repo>:<ref>:<name> will not collide.